### PR TITLE
refactor: card repository

### LINKED ISSUE #1181: [REFACTOR]: card repository
## Summary

Create a separate repository related to card's operations and organize the existing methods of all services.

### Motivation

This change will turn the code more readable.

### PR DESCRIPTION
**Pull request by CatiaAntunes96**
Relates to #1202  #1181 

## Proposed Changes

  - Add card repository
  - Standardisation  of classes and interfaces names of all modules, except votes and comments modules

This pull request closes #1181

### CODE REVIEW

File: backend/src/modules/teams/controller/team.controller.ts [hunk 1]
```diff
@@ -64,9 +64,9 @@ const TeamUser = (permissions: string[]) => SetMetadata('permissions', permissio
 export default class TeamsController {
 	constructor(
 		@Inject(TYPES.applications.CreateTeamApplication)
-		private createTeamApp: CreateTeamApplicationInterface,
+		private createTeamApp: CreateTeamApplication,
```
Comment (nunocaseiro) [inline backend/src/modules/teams/controller/team.controller.ts:67]:
```suggestion
		private createTeamApp: CreateTeamApplicationInterface,
```

File: backend/src/modules/teams/controller/team.controller.ts [hunk 2]
```diff
@@ -64,9 +64,9 @@ const TeamUser = (permissions: string[]) => SetMetadata('permissions', permissio
...
 		@Inject(TYPES.applications.GetTeamApplication)
-		private getTeamApp: GetTeamApplicationInterface,
+		private getTeamApp: GetTeamApplication,
```
Comment (nunocaseiro) [inline backend/src/modules/teams/controller/team.controller.ts:69]:
```suggestion
		private getTeamApp: GetTeamApplicationInterface,
```

**Review by nunocaseiro — CHANGES_REQUESTED** [review]

**Review by nunocaseiro — CHANGES_REQUESTED** [review]

File: backend/src/modules/cards/services/delete.card.service.ts [hunk 3]
```diff
@@ -164,38 +58,46 @@ export default class DeleteCardService implements DeleteCardServiceInterface {
...
-			if (!board) throw Error(UPDATE_FAILED);
-			await session.commitTransaction();
-
-			return board;
+			const boardUpdated = await this.cardRepository.deleteCardFromCardItems(
+				boardId,
+				cardId,
+				cardItemId
+			);
+
```
[the remark this answers is not in the document]
Comment (CatiaAntunes96) [reply backend/src/modules/cards/services/delete.card.service.ts:66]: I added session to all the functions above @nunocaseiro

**Review by nunocaseiro — APPROVED** [review]
LGTM 🚀 gj

**Pull request merged**